testing/sdk_upgrade.py

```diff
@@ -147,6 +147,23 @@ def get_config(package_name, service_name):
     return target_config
 
 
+def update_service(package_name, service_name, additional_options=None, to_package_version=None):
+    update_cmd = ['update', 'start']
+    if to_package_version:
+        update_cmd.append('--package-version={}'.format(to_package_version))
+    if additional_options:
+        with tempfile.NamedTemporaryFile() as opts_f:
+            opts_f.write(json.dumps(additional_options).encode('utf-8'))
+            opts_f.flush()  # ensure json content is available for the CLI to read below
+            sdk_cmd.svc_cli(
+                package_name,
+                service_name,
+                '{} --options={}'.format(' '.join(update_cmd), opts_f.name)
+            )
+    else:
+        sdk_cmd.svc_cli(package_name, service_name, ' '.join(update_cmd), check=True)
```

Comment:
There is a slight difference here between the two commands, ideally we'd want to execute just one. We could: Create the named temporary file and then add `--options=filename` to `update_cmd`, have a single call to `sdk_cmd.svc_cli` and then cleanup the named temporary in a `finally` block. I think a named temporary may be cleaned up when the variable holding it goes out of scope anyway. (I just confirmed in ipython).

Reply:
If I understand your comment correctly, we always create a file handler but choose only to use it if `additional_options` is not None ? Can you put what you mean as a psuedocode ?

Reply:
See the comment here: https://github.com/mesosphere/dcos-commons/pull/2573/files/4b23550c81cdee1457a559057b1be03c18412057#r204701945